openhtf/core/test_executor.py

```diff
@@ -212,8 +198,19 @@ def _execute_test_start(self, phase_exec):
   def _execute_test_teardown(self, phase_exec):
     phase_exec.stop(timeout_s=conf.cancel_timeout_s)
     phase_exec.reset_stop()
-    if self._do_teardown_function and self._teardown_function:
-      outcome = phase_exec.execute_phase(self._teardown_function)
+    try:
+      timeout_s = self._test_options.teardown_function.options.timeout_s
+    except AttributeError:
+      # Force teardown function timeout, otherwise we can hang for a long time
+      # when shutting down, such as in response to a SIGINT.
+      timeout_s = conf.teardown_timeout_s
+
+    teardown_function = (
+        self._test_options.teardown_function and
+        openhtf.PhaseDescriptor.wrap_or_copy(
+            self._test_options.teardown_function, timeout_s=timeout_s))
+    if self._do_teardown_function and teardown_function:
+      outcome = phase_exec.execute_phase(teardown_function)
```

Comment:
Might as well refactor this now:

```python
if self._do_teardown_function and self._test_options.teardown_function:
  teardown_function = openhtf.PhaseDescriptor.wrap_or_copy(...)
  outcome = ...
  ...
```

Reply:
https://github.com/google/openhtf/pull/774